keel-retrofit/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/AuthorizationHeadersInterceptor.kt

```diff
@@ -0,0 +1,32 @@
+package com.netflix.spinnaker.keel.retrofit
+
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator
+import com.netflix.spinnaker.kork.common.Header
+import com.netflix.spinnaker.security.AuthenticatedRequest
+import okhttp3.Interceptor
+import okhttp3.Interceptor.Chain
+import okhttp3.Response
+import org.slf4j.LoggerFactory
+
+/**
+ * Okhttp3 interceptor that adds the X-SPINNAKER-USER and X-SPINNAKER-ACCOUNTS headers to enable authorization
+ * with downstream Spinnaker services.
+ */
+class AuthorizationHeadersInterceptor(private val fiatPermissionEvaluator: FiatPermissionEvaluator) : Interceptor {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun intercept(chain: Chain): Response {
+    var request = chain.request()
+    request.header(Header.USER.header)?.also { user ->
```

Comment:
This interceptor takes care of adding `X-SPINNAKER-ACCOUNTS` based on the `X-SPINNAKER-USER` header already included in the request.

Reply:
very cool